src/pu-utils.c

```diff
@@ -496,3 +506,66 @@ pu_str_pre_remove(gchar *string,
 
     return string;
 }
+
+gboolean
+pu_parse_size(const gchar *string,
```

Comment:
Instead of reinventing the wheel, have you considered using `ped_unit_parse()` from libparted?

Reply:
Unfortunately `ped_unit_parse()` can not be used here as it requires a device of some sort, which is not present when `pu_parse_size()` is used.

Reply:
Yes, it can be used with loop devices. This is the main reason I introduced loop devices here.
The problem here is that we need some way to determine the initial size of the loop device before creating it. And that is where `pu_parse_size()` is used.